### PR TITLE
Deprecate page_number in /search and /find endpoints

### PR DESCRIPTION
**Pull request by jotare**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_models/src/nucliadb_models/search.py [hunk 1]
```diff
@@ -429,17 +429,36 @@ class SearchParamDefaults:
...
+    catalog_page_size = ParamDefault(
         default=20,
         le=200,
         title="Page size",
         description="The number of results to return per page. The maximum number of results per page allowed is 200.",
     )
+    page_number = ParamDefault(
+        default=0,
+        title="Page number",
+        description="The page number of the results to return.\nATENTION: pagination will be deprecated, please, refer to `top_k`",
```
Comment (lferran) [inline nucliadb_models/src/nucliadb_models/search.py:446]:
```suggestion
        description="The page number of the results to return.\nATENTION: pagination is deprecated and this parameter will be removed soon. Please, use `top_k` instead",
```

File: nucliadb_models/src/nucliadb_models/search.py [hunk 2]
```diff
@@ -429,17 +429,36 @@ class SearchParamDefaults:
...
+        deprecated=True,
+    )
+    page_size = ParamDefault(
+        default=20,
+        le=200,
+        title="Page size",
+        description="The number of results to return per page. The maximum number of results per page allowed is 200.\nATENTION: pagination will be deprecated, please, refer to `top_k`",
```
Comment (lferran) [inline nucliadb_models/src/nucliadb_models/search.py:453]:
```suggestion
        description="The number of results to return per page. The maximum number of results per page allowed is 200.\nATENTION: pagination is deprecated and will be removed soon, pleas use to `top_k` instead",
```

File: nucliadb_models/src/nucliadb_models/search.py [hunk 3]
```diff
@@ -647,8 +666,9 @@ class BaseSearchRequest(BaseModel):
         title="Filters",
         description="The list of filters to apply. Filtering examples can be found here: https://docs.nuclia.dev/docs/rag/advanced/search/#filters",  # noqa: E501
     )
-    page_number: int = SearchParamDefaults.page_number.to_pydantic_field()
+    page_number: int = SearchParamDefaults.page_number.to_pydantic_field(deprecated=True)
     page_size: int = SearchParamDefaults.page_size.to_pydantic_field()
```
Comment (lferran) [inline nucliadb_models/src/nucliadb_models/search.py:670]: this one should be deprecated too no?

Comment (jotare) [reply nucliadb_models/src/nucliadb_models/search.py:670]: They are deprecated in the field itself, but I can add it for readability

**Pull request merged**